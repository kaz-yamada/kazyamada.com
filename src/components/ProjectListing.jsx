import React from "react";
import GitHubButton from "react-github-btn";

const ProjectListing = ({ projects }) => {
  return (
    <div className="project-list">
      {projects.map(project => (
        <div className="project" key={project.title}>
          <div className="project-info">
            <h2>
              <a
                className="project-link"
                href={project.source}
                target="_blank"
                rel="noopener noreferrer"
              >
                <div className="project-title">{project.title}</div>
              </a>
            </h2>
            <p>{project.description}</p>
          </div>
          <div className="buttons">
            <GitHubButton
              href={project.source}
              data-size="large"
              data-show-count="true"
            >
              Source
            </GitHubButton>
            {project.path && (
              <a
                className="button"
                href={project.path}
                target="_blank"
                rel="noopener noreferrer"
              >
                View
              </a>
            )}
          </div>
        </div>
      ))}
    </div>
  );
};

export default ProjectListing;
