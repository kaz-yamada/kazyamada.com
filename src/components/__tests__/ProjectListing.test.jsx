import React from "react";

import renderer from "react-test-renderer";

import ProjectListing from "../ProjectListing";

describe("Project Listing", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ProjectListing projects={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
