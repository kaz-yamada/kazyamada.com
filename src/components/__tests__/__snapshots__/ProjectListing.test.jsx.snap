// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Listing renders correctly 1`] = `
<div
  className="project-list"
/>
`;
